strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/ConfigurationManagementServiceImpl.java

```diff
@@ -329,18 +329,18 @@ private void setAllows()
                              {
                                  if (storage.getRepositories() != null && !storage.getRepositories().isEmpty())
                                  {
-                                     for (MutableRepository repository : storage.getRepositories().values())
+                                     for (Repository repository : storage.getRepositories().values())
```

Comment:
Wouldn't it be better to do:
```
for (Repository r : storage.getRepositories().values())
{
    MutableRepository repository = (MutableRepository) repository;
    ...
}
```

And use it like this? If I recall correctly, casting to a sub-class was actually expensive.
(I might be wrong, but it also looks a bit unpleasant).

Reply:
fixed